src/Widgets/HeaderBar.vala

```diff
@@ -90,6 +91,31 @@ public class Camera.Widgets.HeaderBar : Gtk.HeaderBar {
         mode_switch = new Granite.ModeSwitch.from_icon_name (PHOTO_ICON_SYMBOLIC, VIDEO_ICON_SYMBOLIC);
         mode_switch.valign = Gtk.Align.CENTER;
 
+        var mirror_label = new Gtk.Label (_("Horizontal flip")) {
+            hexpand = true,
+            halign = Gtk.Align.START
+        };
+
+        var mirror_switch = new Gtk.Switch ();
+        mirror_switch.activate.connect (() => {
+            request_horizontal_flip ();
+        });
+
+        var mirror_grid = new Gtk.Grid () {
+            hexpand = true
+        };
+        mirror_grid.attach (mirror_label, 0, 0);
+        mirror_grid.attach (mirror_switch, 1, 0);
```

Comment:
Since grid packs horizontal by default, you can use `add` here
```suggestion
        mirror_grid.add (mirror_label);
        mirror_grid.add (mirror_switch);
```
```suggestion
        mirror_grid.attach (mirror_label, 0, 0);
        mirror_grid.attach (mirror_switch, 1, 0);
```